### PR TITLE
Languages can now use their own domains again #1382

### LINKED ISSUE #1382: Multi-domain language setup: non-default language domain 302 redirects to default domain
**Describe the bug**
As in the title

**To Reproduce**
1. Set up two languages in Starterkit and assign a language-specific URL to each
2. In config, set `languages` to true and `languages.detec`t to false (should be the default, anyway)
3. Try to open non-default domain in a browser
4. See it redirecting to the default domain (Chrome network tab shows a 302 redirect)

**Expected behavior**
Needless to say that no redirect should happen unless language detection is active.

**Kirby Version**
3.0.0

**Desktop (please complete the following information):**
 -  macOS
 - chrome, safari

**Additional context**
https://forum.getkirby.com/t/using-different-domains-for-each-language-in-kirby3-does-not-work-for-me-here-need-ideas-to-solve-the-issue/12282

### PR DESCRIPTION
**Pull request by bastianallgeier**
## Describe the PR

After 3.0.0 language routing was no longer possible with custom domains for individual languages. This PR finally fixes this regression. 

## Related issues

- Fixes #1382

## Ready?

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Fixed code style issues with CS fixer and `composer fix`
- [x] Added in-code documentation (if neeed)

### CODE REVIEW

**Comment by distantnative** [comment]
@bastianallgeier "After 3.3.0"?

**Comment by bastianallgeier** [comment]
@distantnative argh, that should be 3.0.0 of course :) I just fixed it.

**Comment by afbora** [comment]
Unfortunately, my first tests failed on my local development
When I entered to `kirby.com`, it needs to be directed to `tr.kirby.com` (browser language), but it is redirected to `en.kirby.com`. Of course if there are no missing in my followingconfig options

**config.php**

````php
return [
    'debug' => true,
    'languages' => true,
    'languages.detect' => true,
];
````

**languages/tr.php**

````php
<?php

return [
    'code' => 'tr',
    'default' => false,
    'direction' => 'ltr',
    'locale' => [
        'tr_TR'
    ],
    'name' => 'Turkish',
    'translations' => [],
    'url' => 'http://tr.kirby.com'
];
````

**languages/en.php**

````php
<?php

return [
    'code' => 'en',
    'default' => true,
    'direction' => 'ltr',
    'locale' => [
        'en_US'
    ],
    'name' => 'English',
    'translations' => [],
    'url' => 'http://en.kirby.com'
];
````

**Comment by afbora** [comment]
In addition, in Turkish (non-default secondary language), pages with Turkish content are redirected to 404 and other pages are redirected to English.

**Comment by bastianallgeier** [comment]
@afbora language detection is a completely separate issue. This fix only takes care about handling different language domains correctly.

**Comment by afbora** [comment]
Ahh, I thought it was solved together :))
Can you take a look at the issue below?

In Turkish frontend (non-default secondary language), pages with Turkish content are redirected to 404 and other pages are redirected to English.

**Entered**: http://tr.kirby.com/notes (No content yet for this language)
**Redirected**: http://en.kirby.com/notes

**Entered**: http://tr.kirby.com/fotograflar (photography.tr.txt exist but opening 404 error page)

````php
return [
    'debug' => true,
    'languages' => true,
    'languages.detect' => false,
];
````

**Comment by bastianallgeier** [comment]
@afbora I think I found it. Could you test again?

**Comment by afbora** [comment]
Now works great for different language patterns (_null, uri, domain_)! ✔️ 

Request for `http://tr.getkirby.com` address as secondary language

| Method | Value |
|:--|:--|
| $kirby->site()->url() | http://tr.getkirby.com |
| $kirby->url() | http://tr.getkirby.com |
| $kirby->language()->url() | http://tr.getkirby.com |
| $kirby->defaultLanguage()->url() | http://en.getkirby.com |
| $kirby->detectedLanguage()->url() | http://tr.getkirby.com |

**Review by afbora — DISMISSED** [review]

File: src/Cms/LanguageRoutes.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+use Kirby\Toolkit\F;
+
+class LanguageRoutes
+{
+    /**
+     * Inject all language routes into
+     * the given routes array
+     *
+     * @param App $kirby
```
Comment (lukasbestle) [inline src/Cms/LanguageRoutes.php]: This needs an absolute class name (same below).

File: src/Cms/LanguageRoutes.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+namespace Kirby\Cms;
+
+use Kirby\Toolkit\F;
+
+class LanguageRoutes
+{
+    /**
+     * Inject all language routes into
+     * the given routes array
```
Comment (lukasbestle) [inline src/Cms/LanguageRoutes.php]: Does it do that? As far as I can see, a new array is created and returned.

File: src/Cms/LanguageRoutes.php [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+     *
+     * @param App $kirby
+     * @return array
+     */
+    public static function create(App $kirby): array
+    {
+        $routes[] = static::home($kirby);
```
Comment (lukasbestle) [inline src/Cms/LanguageRoutes.php:20]: The `$routes` array is not initialized before.

**Review by lukasbestle — COMMENTED** [review]
I love how clean the code is now! 👍👍
My comments are just about a few nitpicky details.

File: src/Cms/LanguageRoutes.php (hunk 1 above)
Comment (bastianallgeier) [reply src/Cms/LanguageRoutes.php]: ✅

File: src/Cms/LanguageRoutes.php (hunk 3 above)
Comment (bastianallgeier) [reply src/Cms/LanguageRoutes.php:20]: ✅

**Pull request merged**